### PR TITLE
refactor: Qdrant implementation

### PR DESCRIPTION
**Pull request by Anush008**
This PR updates the sink implementation to
- Accept a `QdrantClient` object directly instead of host and port to improve the customizability of the client.
- Use the `upload_points()` method which has built-in support for batching and parallelism.
- Use the `collection_exists` API to check for the collection.

Updates the source implementation to
- Accept a `QdrantClient` object directly instead of host and port to improve the customizability of the client.
- Use a paginated scroll implementation.
- Current implementation allows only int point IDs as offset. This fixes it.

### CODE REVIEW

File: src/qdrant/qdrant_datasink.py [hunk 1]
```diff
@@ -1,71 +1,78 @@
...
+        parallel: int = 1,
+        **kwargs: Any,
+    ) -> None:
+        self._collection_name = collection_name
+        self._batch_size = batch_size
+        self._parallel = parallel
+        self._kwargs = kwargs
+
+        # Can't set QdrantClient directly due to serialization issues
+        self._client_opts = client.init_options
```
Comment (Anush008) [inline src/qdrant/qdrant_datasink.py:29]: `init_options` property will be available from the next release of `qdrant-client`. Will have to wait.

**Comment by justinrmiller** [comment]
I need to take a closer look tomorrow but like what I have seen so far. I'll try to get moving on CI for the project.

File: src/qdrant/qdrant_datasink.py (hunk 1 above)
Comment (Anush008) [reply src/qdrant/qdrant_datasink.py:29]: Out now.

Comment (justinrmiller) [reply src/qdrant/qdrant_datasink.py:29]: Great I’ll try it out tonight and merge it if I don’t see any issues. Thanks.

**Comment by justinrmiller** [comment]
Thank you @Anush008! I've noticed an issue with writing datasets built from the Qdrant datasource but I don't believe this PR introduce that issue and I'll follow up on the Ray Slack to try to figure out some next steps.

```
Hello, I'm working on a datasource for Qdrant and seem to be running into an issue with CSV files when I go to write a dataset constructed with the datasource.
(ReadQdrant->Write pid=67656) id: int64
(ReadQdrant->Write pid=67656) payload: struct<id: int64>
(ReadQdrant->Write pid=67656)   child 0, id: int64
(ReadQdrant->Write pid=67656) vector: list<item: double>
(ReadQdrant->Write pid=67656)   child 0, item: double

pyarrow.lib.ArrowInvalid: Unsupported Type:struct<id: int64>
This structure serializes fine in Parquet but errors out on the payload and vector types when I try to write a CSV file. Is it expected that the user pass in some kind of function or schema for CSV to handle these data types?
```

Going to merge this now.

**Pull request merged**